### ISSUE TITLE
HierarchyRequestError when viewing a page that ends with a /

### ISSUE THREAD

**Issue by 2blane** [vuejs/vuepress#2421]
- [x] I confirm that this is an issue rather than a question.

## Bug report

My site which you can view live [here](https://docs.darwincloud.com) has an error whenever I try to visit a page ending in '/'. Let's say you want to visit the link [/cli/](https://docs.darwincloud.com/cli/), instead of being taken to the /cli/ page, an error appears in the console and the home page content is shown on the screen. This only happens in production, and hasn't happened to me locally (hosting on AWS S3 with Cloudfront).
If however, you go to the page [/cli](https://docs.darwincloud.com/cli) without the trailing '/' then it works as expected and the error doesn't appear. Here is the error I get:

![Screen Shot 2020-06-03 at 2 12 17 PM](https://user-images.githubusercontent.com/25189660/83678992-3fb0ea80-a5a4-11ea-8776-4a4ac2022b59.png)

#### Steps to reproduce

View [/cli](https://docs.darwincloud.com/cli) and see how the content appears correctly.

View [/cli/](https://docs.darwincloud.com/cli/) and see how the home page content appears instead of /cli/ and how an error appears in the dev console.

#### What is expected?

There isn't supposed to be an error in the console and the correct page content is supposed to show instead of the home page content.

#### What is actually happening?

An error is occurring and the home page content is showing instead of the content of the correct page.

#### Other relevant information

None

**Comment by 2blane** [comment]
I think this might have to do with Google Analytics?
I found this link:
[https://stackoverflow.com/a/12398221/560145](https://stackoverflow.com/a/12398221/560145)

**Comment by 2blane** [comment]
Turns out, it was a hosting issue. CloudFront was redirecting any page that wasn't found to /index.html like so `/abc/def/` -> `/index.html`. Then, the `/index.html` page was being displayed and throwing an error for some reason - probably because it was confused about whether the page being shown is supposed to be index or /abc/def/.

So, to fix this, I had to make CloudFront route to `index.html` in the current directory. For instance, `/abc/def/` would redirect to `/abc/def/index.html`. This fixed the issue and everything now works as expected.

I guess other hosting providers (GitHub Pages, Firebase, ...) automatically try to find the `index.html` for a subdirectory, but CloudFront doesn't perform any redirection by default. I had to use Lambda Edge in combination with CloudFront to get the redirection to work correctly.
